common/src/main/java/com/wynntils/wynn/model/quests/QuestInfo.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.wynn.model.quests;
+
+import com.wynntils.mc.objects.Location;
+import com.wynntils.mc.utils.ComponentUtils;
+import com.wynntils.utils.Pair;
+import com.wynntils.utils.StringUtils;
+import com.wynntils.wynn.model.CharacterManager;
+import com.wynntils.wynn.objects.ProfessionInfo;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraft.ChatFormatting;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.network.chat.TextComponent;
+
+public class QuestInfo {
+    private static final int NEXT_TASK_MAX_WIDTH = 200;
+    private static final Pattern COORDINATE_PATTERN = Pattern.compile(".*\\[(-?\\d+), ?(-?\\d+), ?(-?\\d+)\\].*");
+
+    private final String name;
+    private final QuestStatus status;
+    private final QuestLength length;
+    private final int level;
+    private String nextTask;
+    /** Additional requirements as pairs of <"profession name", minLevel> */
+    private final List<Pair<String, Integer>> additionalRequirements;
+
+    private final boolean isMiniQuest;
+    private final int pageNumber;
+    private final boolean tracked;
+
+    protected QuestInfo(
+            String name,
+            QuestStatus status,
+            QuestLength length,
+            int level,
+            String nextTask,
+            List<Pair<String, Integer>> additionalRequirements,
+            boolean isMiniQuest,
+            int pageNumber,
+            boolean tracked) {
+        this.name = name;
+        this.status = status;
+        this.length = length;
+        this.level = level;
+        this.nextTask = nextTask;
+        this.additionalRequirements = additionalRequirements;
+        this.isMiniQuest = isMiniQuest;
+        this.pageNumber = pageNumber;
+        this.tracked = tracked;
+    }
+
+    public String getName() {
+        return name;
+    }
```

Comment:
Could this reasonably be made a record?

Reply:
No, since nextTask is not final (and actually `status` should not be either). If we really wanted to, maybe we should split this into like a constant `QuestInfo` with just title, level, etc, and a `QuestProgress` which updates as we progress through a quest. 

It will help the conceptual design, but it will also make us need more code to compensate for the abstraction. I'll have to think about it. So not this PR at least. (But it's likely not the last one in QuestManager)